packages/analytics/amplify_analytics_pinpoint_dart/test/queued_item_store_test.dart

```diff
@@ -0,0 +1,203 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:io';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/dart_queued_item_store.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/queued_item_store.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+import 'package:test/test.dart';
+
+void main() {
+  late DartQueuedItemStore db;
+
+  group('DartQueuedItemStore ', () {
+    setUpAll(() {
+      String? path;
+      if (!zIsWeb) {
+        final tempDir =
+            Directory.systemTemp.createTempSync('amplify_analytics');
+        addTearDown(() => tempDir.deleteSync(recursive: true));
+        path = tempDir.path;
+      } else {
+        addTearDown(() async => await db.clear());
+      }
+      db = DartQueuedItemStore(path);
+    });
+
+    setUp(() async {
+      await db.clear();
+    });
+
+    Future<Iterable<QueuedItem>> getAll() async {
+      return db.getCount(100);
+    }
+
+    test('writes values to storage', () async {
+      const values = ['0', '1', '2', '3', '4', '5'];
+      for (final value in values) {
+        await db.addItem(value);
+      }
+
+      final readItems = await db.getCount(values.length);
+      final readValues = readItems.map((e) => e.value);
+      expect(readValues, equals(values));
+    });
+
+    test('returns first n items in storage', () async {
+      const values = ['0', '1', '2', '3', '4', '5'];
+      for (final value in values) {
+        await db.addItem(value);
+      }
+
+      final readItems = await db.getCount(3);
+      final readValues = readItems.map((e) => e.value);
+      expect(readValues, equals(values.sublist(0, 3)));
+    });
+
+    test(
+        'returns all stored items when get request size exceeds stored item count',
+        () async {
+      const values = ['0', '1', '2', '3', '4', '5'];
+      for (final value in values) {
+        await db.addItem(value);
+      }
+
+      final readItems = await getAll();
```

Comment:
Here you should be explicit since it's what you're testing...

```suggestion
      final readItems = await db.getCount(100);
```

Once you test it, you can take it for granted and use `getAll`